src/last_transactions.test.ts

```diff
@@ -0,0 +1,239 @@
+import * as fs from 'fs';
+import path from 'path';
+import { APIGatewayProxyEvent, Context } from 'aws-lambda';
+import { Client } from 'pg';
+import { DEFAULT_TRANSACTION_COUNT, lastTransactions, lastTransactionsQueryTemplate } from './last-transactions';
+import * as database from './database';
+import * as dapiNames from './dapi-names';
+
+describe('handles http queries for latest transactions for single dataFeedIds', () => {
+  it('responds to a valid query', async () => {
+    const validRows = JSON.parse(
+      fs.readFileSync(path.join(__dirname, '..', 'test', 'fixtures', 'last-transactions-example.json')).toString()
+    );
+
+    const initDb = jest.spyOn(database, 'initDb');
+    const mockQueryImplementation = jest.fn().mockImplementation((_query: string, _parameters: any[]) => ({
+      rows: validRows,
+    }));
+    // @ts-ignore
+    initDb.mockReturnValue({ query: mockQueryImplementation });
+
+    const queryStringParameters = {
+      chainId: '30',
+      beaconId: '0x09a5873667837598bd0990ba2f53d750d545ce435ecdcd44e0b4c64ab7d7d20d',
+    };
+
+    const result = await lastTransactions(
+      {
+        queryStringParameters,
+      } as unknown as APIGatewayProxyEvent,
+      {} as unknown as Context,
+      () => {}
+    );
+    expect(initDb).toHaveBeenCalledTimes(1);
+    expect(mockQueryImplementation).toHaveBeenCalledTimes(1);
+    expect(mockQueryImplementation).toHaveBeenCalledWith(lastTransactionsQueryTemplate, [
```

Comment:
I'd say this is too much detail and would remove that assertion (it also prevents inlining the query which feels more important).